### DISCUSSION TITLE
Any existing pass providers?

### DISCUSSION THREAD

**Discussion by whosnorman** [passes-org/passes#3] [Pass Providers]
Heyo started looking in to playing with this but couldn't find any existing pass providers to make requests to. Brings me two Qs: 1) are there any existing pass providers to use and 2) does every one of my users have to have an app or account with that specific pass provider or is there a way to make a general request that gets fulfilled by any pass provider that the user has?

**Comment by danscan** [comment]
Yo Norm!

> 1) are there any existing pass providers to use

Not yet.

However, there's a devtool that lets you manually enter a result for a given Pass Request using the Polyfill flow, so you can effectively emulate a Pass Provider that supports any given request.

You can try it by going to one of our [examples](https://docs.passes.org/examples/recognizing-users.html), and selecting "Document.Passes" in pane on the right to use the actual `document.passes` ABI, and then sending a request. Just make sure you select "JSON" as the result codec and make the result body valid JSON (we'll make this more friendly soon!)

We're working on a complementary tool that lets you create and send arbitrary Pass Requests to make that interactive debug/dev flow end-to-end. You can preview it at: https://passes.org/request/dev (url may change later when it's released)

> 2) does every one of my users have to have an app or account with that specific pass provider

Nope!

That's the beauty of the protocol. If the requests your app makes are supported by the user's Pass Provider, it will work with your app, and generally your app will be totally agnostic and unaware of which Pass Provider the user uses.

Related to that, we're currently exploring how to support delegating a Pass Request to another Pass Provider to support a world where people may choose to use one primary catch-all Pass Provider for common, everyday requests, and a small set of niche Pass Providers for less common use cases.
